Automattic/sensei
# Use standard approach of displaying filters for list tables

In this PR I'm trying to display filters as it is done for Posts. It helps to avoid issues with positioning of elements. However, I found we have some special cases (the standard approach is to hide filters on mobile, but we have some required filters that we have to display always).

### Known Issues

* I noticed that filters started adding unnecessary params like `_wpnonce` and `_wp_http_referer` to the query. These params we add to the form, the same on the Posts page. But when I submit filters for Posts it doesn't add mentioned params to the query. Can't find how it is filtered there. Any ideas? 🙏 

### Changes proposed in this Pull Request

* Enclose the table, search box and filters into single form.

### Testing instructions

* Try to open Reports, Grading, Students pages.
* Compare them to `trunk`. There might be changes, but they should be "positive" (i.e. it should look like in Pages now instead of "custom" positioning).
* Check them with mobile dimensions.

### Screenshot / Video

*After*
<img width="1606" alt="CleanShot 2022-05-29 at 17 11 32@2x" src="https://user-images.githubusercontent.com/329356/170873535-f2eb91c3-dc1d-4667-8cc8-8ad30ae80b4c.png">
<img width="290" alt="CleanShot 2022-05-29 at 17 11 51@2x" src="https://user-images.githubusercontent.com/329356/170873546-cbc6fad4-28ba-4fa3-9242-83474e491eb0.png">
<img width="1280" alt="CleanShot 2022-05-24 at 10 52 01@2x" src="https://user-images.githubusercontent.com/329356/169979189-1546191e-61bc-4f8d-80e7-f0d32885a415.png">
<img width="319" alt="CleanShot 2022-05-24 at 10 52 19@2x" src="https://user-images.githubusercontent.com/329356/169979220-935d54a6-6201-4daa-862e-a8c3cff8c0ac.png">
<img width="1280" alt="CleanShot 2022-05-24 at 10 52 47@2x" src="https://user-images.githubusercontent.com/329356/169979246-6a1ca838-8d45-4013-85c1-601361dc8fc0.png">
<img width="322" alt="CleanShot 2022-05-24 at 10 53 02@2x" src="https://user-images.githubusercontent.com/329356/169979310-c80efde8-1eff-4030-8b1f-23cec7d3aa23.png">
<img width="806" alt="CleanShot 2022-05-29 at 17 29 55@2x" src="https://user-images.githubusercontent.com/329356/170874557-f319fce8-d3d6-44ac-bcc9-f19d423d05df.png">


*Before*
<img width="1279" alt="CleanShot 2022-05-24 at 10 56 34@2x" src="https://user-images.githubusercontent.com/329356/169980128-d7e56c08-488f-46b2-98f5-fe466245b3bc.png">
<img width="320" alt="CleanShot 2022-05-24 at 10 56 48@2x" src="https://user-images.githubusercontent.com/329356/169980206-4981cb6b-cd53-4012-9c1a-ff03692c39f5.png">
<img width="1280" alt="CleanShot 2022-05-24 at 10 57 11@2x" src="https://user-images.githubusercontent.com/329356/169980243-b0c00253-fa2f-48e1-a357-268ca4dcdffc.png">
<img width="321" alt="CleanShot 2022-05-24 at 10 57 26@2x" src="https://user-images.githubusercontent.com/329356/169980325-5e813586-3cde-4812-9147-ac9e2da0d41c.png">
<img width="1280" alt="CleanShot 2022-05-24 at 10 57 52@2x" src="https://user-images.githubusercontent.com/329356/169980372-95fb26e9-fb90-4e62-9f77-1698425e9e80.png">
<img width="321" alt="CleanShot 2022-05-24 at 10 58 08@2x" src="https://user-images.githubusercontent.com/329356/169980422-da52bd64-edad-4336-8b6d-30f934177e72.png">
